src/systems/world_exporter/WorldExporter.cc

```diff
@@ -0,0 +1,382 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <ignition/plugin/Register.hh>
+
+#include <ignition/gazebo/components/Geometry.hh>
+#include <ignition/gazebo/components/Link.hh>
+#include <ignition/gazebo/components/Material.hh>
+#include <ignition/gazebo/components/Model.hh>
+#include <ignition/gazebo/components/Name.hh>
+#include <ignition/gazebo/components/ParentEntity.hh>
+#include <ignition/gazebo/components/Pose.hh>
+#include <ignition/gazebo/components/Transparency.hh>
+#include <ignition/gazebo/components/Visual.hh>
+#include <ignition/gazebo/components/World.hh>
+#include <ignition/gazebo/EntityComponentManager.hh>
+#include <ignition/gazebo/Util.hh>
+
+#include <sdf/Visual.hh>
+#include <sdf/Mesh.hh>
+#include <sdf/Model.hh>
+
+#include <ignition/common/Material.hh>
+#include <ignition/common/MeshManager.hh>
+#include <ignition/common/Mesh.hh>
+#include <ignition/common/SubMesh.hh>
+
+#include <ignition/math/Matrix4.hh>
+
+#include <ignition/common/ColladaExporter.hh>
+
+#include "WorldExporter.hh"
+
+using namespace ignition;
+using namespace gazebo;
+using namespace systems;
+
+
+class ignition::gazebo::systems::WorldExporterPrivate
+{
+  private:
+    bool rendered{false};
+    common::Mesh worldMesh;
+    Entity world;
+    std::map<Entity, std::pair<math::Pose3d, Entity>> poses;
+    std::vector<math::Matrix4d> subMeshMatrix;
```

Comment:
Do these variables need to be member variables? Many of them only seem to be used within the scope of the `Export` function, I think they could just be defined there.